### DISCUSSION TITLE
Segmentation faults on diann-1.8 using linux command line, speclib generation step

### DISCUSSION THREAD

**Discussion by jk3755** [vdemichev/DiaNN#270] [Q&A]
Hi Vadim -

I have been using diann-1.8 on our lab server segmentation fault errors during the spectral library generation step. I have 4 fasta files that are used individually to generate 4 different spectral libraries, 3 of them seem to work fine most of the time, but the 4th (which is the UP000005640_9606.fasta human proteome file) gives me a segmentation fault every time. 

I first attempted a workaround by splitting the fasta file in half and then running the analysis (including the speclib generation step) on each half of the fasta file at a time. I did not get a segmentation fault for the first fasta, but did get segmentation fault on the second fasta. 

I don't think this is a running out of memory issue, as our lab server has 256gb, but in any case after reading through the manual a bit I then attempted to run the spectral library generation step separately from the analysis, and I also ran it stepwise with --min-pr-mz and --max-pr-mz set to intervals of 100 at a time (so first command would be diann-1.8 --min-pr-mz 300 --max-pr-mz 400 ..., and so on until reaching 1700 and 1800). When running like this, all intervals successfully generated spectral libraries except for the pr-mz ranges of 800-900 and 1300-1400

Below is the bash code used to run the previous commands:
PARAMS_LIB="--lib "" --threads 16 --verbose 4 --qvalue 0.01 --matrices --gen-spec-lib --predictor --fasta-search --min-fr-mz 200 --max-fr-mz 1800 --met-excision --cut K*,R* --missed-cleavages 2 --min-pep-len 7 --max-pep-len 30 --min-pr-charge 1 --max-pr-charge 4 --unimod4 --var-mods 5 --var-mod UniMod:35,15.994915,M --var-mod UniMod:1,42.010565,*n --monitor-mod UniMod:1 --mass-acc 10 --mass-acc-ms1 20 --reanalyse --smart-profiling --global-norm"
fasta_coding_full="/home/kesner/proteomics/JK1153/reference/UP000005640_9606.fasta"
for X in {3..17} ; do A=$((X*100)); B=$(((X+1)*100)); OUT="/home/kesner/proteomics/JK1153/speclib/coding/coding_$A$B"; diann-1.8 --fasta $fasta_coding_full --out $OUT --out-lib $OUT --min-pr-mz $A --max-pr-mz $B $PARAMS_LIB; done

Thanks for any help/advice and please let me know if you need any more info from me

Best,
Jordan

**Comment by vdemichev** [comment]
Hi Jordan,

Thank you so much for troubleshooting this! Could you please share the minimal FASTA file you have so far which produces a segfault? 

Best,
Vadim

**Reply by jk3755** [reply]
No problem! Attached is the fasta file that produces the segfault. See also the attached image for output in the console

Console output:
![Capture](https://user-images.githubusercontent.com/37669924/145865955-ea83f76b-046e-4bbb-bf62-1e93eaa2820d.PNG)

Fasta file:
[UP000005640_9606_part2.txt](https://github.com/vdemichev/DiaNN/files/7705940/UP000005640_9606_part2.txt)

Command which produces the segfault:
diann-1.8 --lib "" --out "/home/kesner/proteomics/JK1153/testing/test" --outlib "/home/kesner/proteomics/JK1153/testing/test" --fasta "/home/kesner/proteomics/JK1153/testing/UP000005640_9606_part2.fasta" --threads 32 --verbose 4 --qvalue 0.01 --matrices --gen-spec-lib --predictor --fasta-search --min-fr-mz 200 --max-fr-mz 1800 --met-excision --cut K*,R* --missed-cleavages 2 --min-pep-len 7 --max-pep-len 30 --min-pr-charge 1 --max-pr-charge 4 --unimod4 --var-mods 5 --var-mod UniMod:35,15.994915,M --var-mod UniMod:1,42.010565,*n --monitor-mod UniMod:1 --mass-acc 10 --mass-acc-ms1 20 --min-pr-mz 300 --max-pr-mz 1800 --reanalyse --smart-profiling --global-norm
